src/main/java/software/amazon/awssdk/crt/http/Http2ClientConnection.java

```diff
@@ -65,15 +68,19 @@ public Http2ClientConnection(long connectionBinding) {
      * @return When this future completes without exception, the peer has
      *         acknowledged the settings and the change has been applied.
      */
-    public CompletableFuture<Void> changeSettings(final Http2ConnectionSetting settings[]) {
-        throw new CrtRuntimeException("Unimplemented");
+    public CompletableFuture<Void> changeSettings(final List<Http2ConnectionSetting> settings) {
+        CompletableFuture<Void> completionFuture = new CompletableFuture<>();
+        http2ClientConnectionChangeSettings(getNativeHandle(), completionFuture,
```

Comment:
Can’t pass futures to JNI and support Android, also this is a coupling we don’t need. I understand if for consistency, I guess, but this should be a functor object, not a future. You can always have a futures api that sits on top of that.

Reply:
Oh, I see the code around uses `AsyncCallback` class. The signer is the only place we use Future directly. I can use the `AysncCallback` class for sure

Reply:
https://github.com/awslabs/aws-crt-java/pull/367/files#diff-210ac8ee910f94aacc1e516387a4a2304b24cd5787216d51add1486a9e736a37R62 Seems like we are targetting 24 as minimal requirement for andriod. So, should be good?